scripts/google_cloud_storage.py

```diff
@@ -26,7 +26,13 @@ def upload_files(
 
     def get_files(self, from_date, output_dir):
 
-        for blob in self.client.list_blobs(self.bucket_name):
+        blobs = [
+            blob
+            for blob in self.client.list_blobs(self.bucket_name)
+            if not blob.name.endswith("/")
+        ]
```

Comment:
I think it was fine as it was, but if you are using a list comprehension than the date comparison should be done here.

Reply:
Also, you should also use a generator instead of a list comp in this scenario, as at the moment, you are looping over most of the matching elements twice.

Reply:
I've just added date comparison conditions to list comprehension but will make it generator now, thanks!

Reply:
Ok, added generator now, thanks!

Reply:
Generator removed now as requested👍